### ISSUE TITLE
Replace `if s.split('.').count() == 1` with `if s.contains('.')`

### ISSUE THREAD

**Issue by Logarithmus** [rust-lang/rust-clippy#9158]
### What it does

I've found this "ingenious" snippet in a big Rust codebase, and Clippy didn't catch it.

### Lint Name

str-split-count-1

### Example

```rust
if url.split('|').count() == 1 {

}
```

Could be written as:

```rust
if url.contains('.') {

}
```

**Comment by klensy** [comment]
But this not equal code? In first example you want exactly one separator, where in second you want >= 1.

**Comment by Logarithmus** [comment]
@klensy sorry, I forgot `!` before the second snippet. Now it should do the same thing perhaps?

**Comment by klensy** [comment]
>but exactly one part of a string, which is possible iff no separators found at all.

Ah, yes.

**Comment by smoelius** [comment]
In the example, `split('|')` should be `split('.')`, I think(?).

**Comment by lukaslueg** [comment]
Can this be generalized to `if the result of std::iter::Iterator::count() is compared to 1 or 0, it can *always* be replaced by *some* variant of std::iter::Iterator::any()`?
